pkg/controller/istio/istio_controller_test.go

```diff
@@ -303,4 +309,477 @@ var _ = Describe("Istio controller tests", func() {
+			customDSCP := numorstring.DSCPFromInt(10)
+			istioCustomDSCP := &operatorv1.Istio{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "default",
+				},
+				Spec: operatorv1.IstioSpec{
+					DSCPMark: &customDSCP,
+				},
+			}
+			Expect(cli.Create(ctx, installation)).NotTo(HaveOccurred())
+			Expect(cli.Create(ctx, istioCustomDSCP)).NotTo(HaveOccurred())
+
+			r := &ReconcileIstio{
+				Client:   cli,
+				scheme:   scheme,
+				provider: operatorv1.ProviderNone,
+				status:   mockStatus,
+			}
+
+			_, err := r.Reconcile(ctx, reconcile.Request{NamespacedName: types.NamespacedName{Name: "default"}})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// Verify DSCPMark was preserved
+			updatedIstio := &operatorv1.Istio{}
+			Expect(cli.Get(ctx, types.NamespacedName{Name: "default"}, updatedIstio)).NotTo(HaveOccurred())
+			Expect(updatedIstio.Spec.DSCPMark).NotTo(BeNil())
+			Expect(updatedIstio.Spec.DSCPMark.ToUint8()).To(Equal(uint8(10)))
+		})
+	})
+
+	Context("FelixConfiguration patching tests", func() {
+		BeforeEach(func() {
+			createResources()
+		})
+
+		It("should patch FelixConfiguration with IstioAmbientMode", func() {
+			fc := &crdv1.FelixConfiguration{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "default",
+				},
+			}
+			Expect(cli.Create(ctx, fc)).NotTo(HaveOccurred())
+
+			r := &ReconcileIstio{
+				Client:   cli,
+				scheme:   scheme,
+				provider: operatorv1.ProviderNone,
+				status:   mockStatus,
+			}
+
+			_, err := r.Reconcile(ctx, reconcile.Request{NamespacedName: types.NamespacedName{Name: "default"}})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// Verify FelixConfiguration was patched
+			updatedFC := &crdv1.FelixConfiguration{}
+			Expect(cli.Get(ctx, types.NamespacedName{Name: "default"}, updatedFC)).NotTo(HaveOccurred())
+			Expect(updatedFC.Spec.IstioAmbientMode).NotTo(BeNil())
+			Expect(*updatedFC.Spec.IstioAmbientMode).To(Equal("Enabled"))
+			Expect(updatedFC.Annotations).To(HaveKey(istio.IstioOperatorAnnotationMode))
+			Expect(updatedFC.Annotations[istio.IstioOperatorAnnotationMode]).To(Equal("Enabled"))
+		})
+
+		It("should patch FelixConfiguration with IstioDSCPMark", func() {
+			fc := &crdv1.FelixConfiguration{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "default",
+				},
+			}
+			Expect(cli.Create(ctx, fc)).NotTo(HaveOccurred())
+
+			dscpMark := numorstring.DSCPFromInt(23)
+			istioCR.Spec.DSCPMark = &dscpMark
+			Expect(cli.Update(ctx, istioCR)).NotTo(HaveOccurred())
+
+			r := &ReconcileIstio{
+				Client:   cli,
+				scheme:   scheme,
+				provider: operatorv1.ProviderNone,
+				status:   mockStatus,
+			}
+
+			_, err := r.Reconcile(ctx, reconcile.Request{NamespacedName: types.NamespacedName{Name: "default"}})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// Verify FelixConfiguration was patched with DSCP mark
+			updatedFC := &crdv1.FelixConfiguration{}
+			Expect(cli.Get(ctx, types.NamespacedName{Name: "default"}, updatedFC)).NotTo(HaveOccurred())
```

Comment:
```suggestion
```

I think we can just do a single test to verify default values are specified? Avoids needing all of this duplication.